packages/shared/src/hooks/streaks/useStreakRecover.ts

```diff
@@ -78,6 +74,22 @@ export const useStreakRecover = ({
     });
   }, [updateAlerts]);
 
+  const { data, isLoading } = useQuery<StreakQueryData>({
+    queryKey: generateQueryKey(RequestKey.UserStreakRecover),
+    queryFn: async () => {
+      const res = await gqlClient.request(USER_STREAK_RECOVER_QUERY);
+
+      if (!res?.streakRecover?.canRecover) {
```

Comment:
Oh wait doesn't this also return false if in general you can't recover?
(so not if you didn't just loose it?)

I feel this should only trigger on a user action right?

Reply:
I can put a check on query string put from the notification. This exist only if user clicked. What do you think?

Reply:
Updated ✅